aip/general/0193.md

```diff
@@ -34,272 +34,154 @@ is necessary, you **should** provide a link where a reader can get more
+The error domain  typically the registered service name of the tool or product
+that  generated the error. The domain must be a globally unique value, and is
+typically the service name, e.g. `pubsub.googleapis.com`.
 
-This practice is critical so that machine actors do not need to rely on
-`LocalizedMessage.message`, which is subject to change and is not part
-of the API contract.
+The (reason, domain) pair form a machine-readable way of identifying a particular error.
+Services **must** use the same (reason, domain) pair for the same error, and
+**must not** use the same (reason, domain) pair for logically different errors.
+The decision about whether two errors are "the same" or not is not always clear, but
+should generally be considered in terms of the expected action a client might take
+to resolve them.
 
-Consider the following example:
+The `metadata` field is a map of key/value pairs providing additional
+dynamic information as context. Each key within `metadata` **must** be at most
+64 characters long, and conform to the regular expression `[a-z][a-zA-Z0-9-_]+`.
 
-> The Book, "The Great Gatsby", is unavailable at the Library, "Garfield
-> East".  It is expected to be available again on 2199-05-13.
+Any request-specific information which contributes to the `Status.message` or
+`LocalizedMessage.message` messages **must** be represented within `metadata`.
+This practice is critical so that machine actors do not need to parse error
+messages to extract information.
 
-The preceding error message is made actionable by the context, both
-originating from the request, the title of the Book, the name of the
-Library, and by the information that is known only by the service, that
-is, the expected return date of the Book.
+For example consider the following message:
 
-All dynamic variables found in error messages **must** also be present
-in the `map<string, string>`, `ErrorInfo.metadata` (found on the
-*required* `ErrorInfo`). For example, the `metadata` map for the sample
-error message above will include *at least* the following key/value
-pairs:
+> An &lt;e2-medium&gt; VM instance with &lt;local-ssd=3,nvidia-t4=2&gt; is currently unavailable
+> in the &lt;us-east1-a&gt; zone. Consider trying your request in the &lt;us-central1-f,us-central1-c&gt;
+> zone(s), which currently has/have capacity to accommodate your request. Alternatively,
+> you can try your request again with a different VM hardware configuration
+> or at a later time. For more information, see the troubleshooting documentation.
 
-```
-bookTitle: "The Great Gatsby"
-library: "Garfield East"
-expectedReturnDate: "2199-05-13"
-```
+The `ErrorInfo.metadata` map for the same error could be:
 
-The following example shows an additional example of a metadata map for
-information about virtual machines:
+- "zone": "us-east1-a",
+- "vmType": "e2-medium",
+- "attachment": "local-ssd=3,nvidia-t4=2",
+- "zonesWithCapacity": "us-central1-f,us-central1-c"
```

Comment:
Should these be formatted as code? I guess wondering why it was switched to bulleted list even though it is a "code examle"?

Reply:
I'm not sure... it's not *really* code - they're name/value pairs. (I replaced the example entirely, as I thought it was unrealistic - if we're going to use books, I wouldn't expect a title, but a resource name... I figured using the same example that we have for the complete error would be better.)

The commas were a mistake though.

I've reformatted each entry in the map as code - we could reformat it all as a JSON object if you'd like?